### PR TITLE
Client - Message children v2

### LINKED ISSUE #297: Client - update get_message_children to v2
## Description

The message children endpoint has been changed
- update restful API route to v2
- update the response format
- `test_message_children` tests should be updated and passed

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

Fixes #297 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

test_message_children.c

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/restful/get_message_children.c [hunk 1]
```diff
@@ -95,40 +96,34 @@ int deser_msg_children(char const *const j_str, res_msg_children_t *res) {
...
-
-    // message ID
-    if ((ret = json_get_string(data_obj, JSON_KEY_MSG_ID, res->u.data->msg_id, sizeof(res->u.data->msg_id))) != 0) {
-      printf("[%s:%d]: parsing %s failed\n", __func__, __LINE__, JSON_KEY_MSG_ID);
-      goto end;
-    }
+  // allocate message children object
+  res->u.data = msg_children_new();
+  if (res->u.data == NULL) {
+    printf("[%s:%d]: msg_children_t object allocation filaed\n", __func__, __LINE__);
```
Comment (matjazv) [inline src/client/api/restful/get_message_children.c]: Typo in a word. `filaed` -> `failed`

File: src/client/api/restful/get_message_children.c [hunk 2]
```diff
@@ -185,3 +180,28 @@ int get_message_children(iota_client_conf_t const *ctx, char const msg_id[], res
...
+    printf("%s\tmessageId: %s\n", PRINT_INDENTATION(indentation), data->msg_id);
+    printf("%s\tmaxResults: %d\n", PRINT_INDENTATION(indentation), data->max_results);
+    printf("%s\tcount: %d\n", PRINT_INDENTATION(indentation), data->count);
+    printf("%s\tchildrenMessageIds: [\n", PRINT_INDENTATION(indentation));
+    int len = utarray_len(data->children);
+    for (int i = 0; i < len; i++) {
+      printf(i > 0 ? ",\n" : "");
+      printf("%s\t\t%s", PRINT_INDENTATION(indentation), *(char **)utarray_eltptr(data->children, i));
+    }
+    printf("\n");
```
Comment (matjazv) [inline src/client/api/restful/get_message_children.c]: There will be an empty line in empty array if there are no children message IDs.
Maybe you could also add an unit test where there is no children message IDs inside a JSON result?

**Review by matjazv — CHANGES_REQUESTED** [review]

**Review by oopsmonk — APPROVED** [review]

File: src/client/api/restful/get_message_children.c (hunk 1 above)
Comment (rebinmannady93) [reply src/client/api/restful/get_message_children.c]: Done.

File: src/client/api/restful/get_message_children.c (hunk 2 above)
Comment (rebinmannady93) [reply src/client/api/restful/get_message_children.c]: Done, handles case without children.

**Review by matjazv — APPROVED** [review]

**Pull request merged**